src/core/drive/page_view.js

```diff
@@ -16,7 +16,7 @@ export class PageView extends View {
   }
 
   renderPage(snapshot, isPreview = false, willRender = true, visit) {
-    const shouldMorphPage = this.isPageRefresh(visit) && this.snapshot.shouldMorphPage
+    const shouldMorphPage = this.isPageRefresh(visit) && snapshot.shouldMorphPage && this.snapshot.shouldMorphPage
```

Comment:
@scuml thank you for opening this PR.

Conceptually, I wonder if we should be checking the current snapshot at all. For example, when determining whether to initiate a [Visit or a Full Reload](https://turbo.hotwired.dev/handbook/drive#ensuring-specific-pages-trigger-a-full-reload), The `PageRenderer` only scans the _new_ snapshot head for a `<meta>` element:

* [PageRenderer.shouldRender](https://github.com/hotwired/turbo/blob/00527e50ed1a4298ce52f968f2a007f6cfd30776/src/core/drive/page_renderer.js#L14)
* [PageRenderer.reloadReason](https://github.com/hotwired/turbo/blob/00527e50ed1a4298ce52f968f2a007f6cfd30776/src/core/drive/page_renderer.js#L18)

Similarly, when using the same `<meta>` to determine whether to ["Break out" of a Frame](https://turbo.hotwired.dev/handbook/frames#“breaking-out”-from-a-frame), the `FrameController` scans the _new_ snapshot:

* [FrameController.loadResponse](https://github.com/hotwired/turbo/blob/00527e50ed1a4298ce52f968f2a007f6cfd30776/src/core/frames/frame_controller.js#L136)

@afcapel @jorgemanrubia I wonder, is this sufficient?

```suggestion
    const shouldMorphPage = this.isPageRefresh(visit) && snapshot.shouldMorphPage
```

Reply:
Interesting you mention that, as that was my initial implementation: https://github.com/hotwired/turbo/pull/1123/commits/c3eacbc25be112adf8663f4c662dcf87bdcc5bcf

I put the other check in as a safety if I missed a use-case - but thinking of it along the lines of "breaking out of a frame", I think you are right that the 'current page check' can be removed.  

This would also offer the user an extra degree of freedom where they can now opt-out and opt-back in to a morph render.